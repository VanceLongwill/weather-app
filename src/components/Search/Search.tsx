import * as React from 'react'

import { Button, Input } from 'antd'
import styled from 'styled-components'

const SearchContainer = styled.div`
  padding: 10px;
  width: 900px;
  display: flex;
`

interface ISearchProps {
  handleSubmit: (query: string) => {}
  loading: boolean
}

interface ISearchState {
  query: string
}

export class Search extends React.Component<ISearchProps, ISearchState> {
  public readonly state: ISearchState

  constructor(props: ISearchProps) {
    super(props)
    this.state = {
      query: '',
    }
  }

  public handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    const query = e.target.value
    this.setState(prevState => ({ ...prevState, query }))
  }

  public handleSearchButtonPress = () => {
    const { handleSubmit } = this.props
    const { query } = this.state
    this.setState(prevState => ({ ...prevState, query: '' }))
    handleSubmit(query)
  }

  public render() {
    const { query } = this.state
    const { loading } = this.props
    return (
      <SearchContainer>
        <Input type="text" value={query} onChange={this.handleChange} />
        <Button
          disabled={loading}
          type="primary"
          onClick={this.handleSearchButtonPress}>
          Search
        </Button>
      </SearchContainer>
    )
  }
}
