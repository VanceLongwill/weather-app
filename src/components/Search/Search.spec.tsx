import { shallow } from 'enzyme'
import * as React from 'react'

import { Search } from './Search'

describe('Search', () => {
  it('should match snapshot', () => {
    const component = shallow(
      <Search handleSubmit={jest.fn()} loading={false} />
    )
    expect(component).toMatchSnapshot()
  })

  it('button should be disabled while loading', () => {
    const component = shallow(
      <Search handleSubmit={jest.fn()} loading={true} />
    )

    expect(component.find('Button').prop('disabled')).toBe(true)
  })

  it('should call handleSubmit when button is pressed', () => {
    const spy = jest.fn()
    const component = shallow(<Search handleSubmit={spy} loading={false} />)
    component.find('Button').simulate('click')
    expect(spy.mock.calls.length).toBe(1)
  })
})
