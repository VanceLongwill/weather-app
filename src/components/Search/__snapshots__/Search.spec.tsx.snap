// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should match snapshot 1`] = `
<Search__SearchContainer>
  <Search__Input
    dataSource={Array []}
    onChange={[Function]}
    placeholder="New York, US"
    value=""
  />
  <Button
    block={false}
    disabled={false}
    ghost={false}
    htmlType="button"
    loading={false}
    onClick={[Function]}
    type="primary"
  >
    Search
  </Button>
</Search__SearchContainer>
`;
